packages/collector/test/nativeModuleRetry/test.js

```diff
@@ -6,15 +6,49 @@
 'use strict';
 
 const { expect } = require('chai');
-const fs = require('node:fs/promises');
+const fsPromise = require('node:fs/promises');
 const os = require('os');
 const path = require('path');
+const fs = require('fs');
 
 const config = require('../../../core/test/config');
 const { retry } = require('../../../core/test/test_util');
 const ProcessControls = require('../test_util/ProcessControls');
 const globalAgent = require('../globalAgent');
 
+/**
+ * Test suite for verifying the fallback mechanism for loading native add-ons.
+ *
+ * This test ensures that if a native add-on (event-loop-stats or gcstats.js) cannot be loaded initially,
+ * the system correctly falls back to using precompiled versions of these add-ons. This ensures the application
+ * functions as expected even if the native module is missing or cannot be compiled.
+ *
+ * **Issue with npm Workspaces**:
+ *
+ * With `npm workspaces`, dependencies are installed in the root `node_modules` folder instead of individual
+ * package `node_modules` folders. This can complicate managing native modules and their precompiled binaries,
+ * as the fallback mechanism must copy to the correct path. This test verifying that precompiled binaries are copied
+ * and loaded correctly to the @instana/shared-metrics/node_modules folder.
+ *
+ * The test operates as follows:
+ *
+ * 1. **Setup**:
+ *      - Temporarily moves the installed native module to a backup directory to simulate its absence.
```

Comment:
from the .... root node_modules? And why do we do that?

Reply:
npm workspaces installs event-loop-stats on the root node_modules folder.

Reply:
Yes. Why do we have to move the root modules dependencies if we operate on shared metrics node_modules?